mvlearn/embed/dcca.py

```diff
@@ -0,0 +1,349 @@
+                and epoch < self.epoch_num:
+            self.model.train()
+            batch_idxs = list(BatchSampler(RandomSampler(range(data_size)),
+                                           batch_size=self.batch_size,
+                                           drop_last=False))
+            current_loss = train_loss
+            for batch_idx in batch_idxs:
+                self.optimizer.zero_grad()
+                batch_x1 = x1[batch_idx, :]
+                batch_x2 = x2[batch_idx, :]
+                o1, o2 = self.model(batch_x1, batch_x2)
+                loss = self.loss(o1, o2)
+                train_losses.append(loss.item())
+                loss.backward()
+                self.optimizer.step()
+            train_loss = np.mean(train_losses)
+            if self.print_train_log_info:
+                info_string = "Epoch {:d}/{:d},"\
+                    " training_loss: {:.4f}"
+                print(info_string.format(epoch + 1, self.epoch_num,
+                      train_loss))
+
+            torch.save(self.model.state_dict(), checkpoint)
+            epoch += 1
+
+        # train_linear_cca
+        if self.linear_cca is not None:
+            losses, outputs = self._get_outputs(x1, x2)
+            self._train_linear_cca(outputs[0], outputs[1])
+
+        checkpoint_ = torch.load(checkpoint)
+        self.model.load_state_dict(checkpoint_)
+
+        self.is_fit = True
+        return self
+
+    def transform(self, Xs, return_loss=False):
+        r"""
+        Embeds data matrix(s) using the trained deep networks and fitted CCA
+        projection matrices. May be used for out-of-sample embeddings.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+            A list of data matrices from each view to transform based on the
+            prior fit function. If view_idx defined, then Xs is a 2D data
+            matrix corresponding to a single view.
+
+        Returns
+        -------
+        Xs_transformed : list of array-likes or array-like
+            Transformed samples. Same structure as Xs, but potentially
+            different n_features_i.
+        loss : float
+            Average loss over data, defined as negative correlation of
+            transformed views. Only returned if ``return_loss=True``.
+        """
+
+        if not self.is_fit:
+            raise RuntimeError("Must call fit function before transform")
+        Xs = check_Xs(Xs, multiview=True)
+        x1 = torch.DoubleTensor(Xs[0])
+        x2 = torch.DoubleTensor(Xs[1])
+
+        with torch.no_grad():
+            losses, outputs = self._get_outputs(x1, x2)
+            outputs = self.linear_cca.test(outputs[0], outputs[1])
+            if return_loss:
+                return outputs, np.mean(losses)
+            return outputs
+
+    def _train_linear_cca(self, x1, x2):
```

Comment:
Is this function necessary if it's only one line? You also need to document x1 and x2